src/rtctools_heat_network/workflows/goals/minimize_tco_goal.py

```diff
@@ -1,69 +1,149 @@
         self,
-        priority=2,
-        number_of_years=25.0,
+        priority: int = 2,
+        number_of_years: float = 25.0,
+        custom_asset_type_maps: Optional[Dict[str, Set[str]]] = None,
     ):
+        """
+        Initialize the MinimizeTCO goal.
+
+        Args:
+            priority (int): The priority of this goal.
+            number_of_years (float): The number of years over which to calculate the costs.
+        """
         self.priority = priority
         self.number_of_years = number_of_years
+        self.function_nominal = 1.0e6
+
+        default_asset_type_maps = {
+            "operational": {"source", "ates"},
+            "fixed_operational": {"source", "ates", "buffer"},
+            "investment": {
+                "source",
+                "ates",
+                "buffer",
+                "demand",
+                "heat_exchanger",
+                "heat_pump",
+                "pipe",
+            },
+            "installation": {
+                "source",
+                "ates",
+                "buffer",
+                "demand",
+                "heat_exchanger",
+                "heat_pump",
+                "pipe",
+            },
+            "annualized": {
+                "source",
+                "ates",
+                "buffer",
+                "demand",
+                "heat_exchanger",
+                "heat_pump",
+                "pipe",
+            },
+        }
+
+        self.asset_type_maps = (
+            custom_asset_type_maps
+            if custom_asset_type_maps is not None
+            else default_asset_type_maps
+        )
+
+    def _calculate_cost(
+        self,
+        optimization_problem: TechnoEconomicMixin,
+        cost_type,
+        asset_types: Set[str],
+        cost_type_map: Dict[str, float],
+        options: Dict[str, Any],
+    ) -> MX:
+        """
+        Calculate the cost for given asset types using a specified cost map.
 
-    def function(self, optimization_problem: TechnoEconomicMixin, ensemble_member):
+        Args:
+            optimization_problem (TechnoEconomicMixin): The optimization problem instance.
+            cost_type (str): The type of cost to calculate ("operational",
+                            "fixed_operational", "investment", "installation")
+            asset_types (Set[str]): Set of asset types to consider for cost calculation.
+            cost_type_map (Dict[str, Any]): Mapping of assets to their respective costs.
+            options (Dict[str, Any]): Options dictionary from heat_network_options
+
+        Returns:
+            MX object: CasADi expression with total cost for the given asset types.
+        """
         obj = 0.0
+        for asset_type in asset_types:
+            for asset in optimization_problem.heat_network_components.get(asset_type, []):
+                extra_var = optimization_problem.extra_variable(cost_type_map[asset])
+                if options["discounted_annualized_cost"]:
```

Comment:
This if statement logic is not super intuitive, maybe add a comment saying that we only want the operational cost for a single year when we use annualized capex.